mantle/cmd/ore/aliyun/make-public.go

```diff
@@ -0,0 +1,79 @@
+// Copyright 2021 Red Hat
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package aliyun
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/spf13/cobra"
+)
+
+var (
+	cmdPublic = &cobra.Command{
+		Use:   "make-public <image:region...>",
+		Short: "Mark images on aliyun publicly available",
+		Long:  `Mark one or more images on aliyun publicly available`,
+		RunE:  makePublic,
+
+		SilenceUsage: true,
+	}
+)
+
+func init() {
+	Aliyun.AddCommand(cmdPublic)
+}
+
+func makePublic(cmd *cobra.Command, args []string) error {
+	if len(args) == 0 {
+		return fmt.Errorf("Specify one ore more image:region pairs.\n")
+	}
+
+	for _, pair := range args {
+		if strings.Contains(pair, ":") == false {
+			return fmt.Errorf("Argument isn't a valid image:region pair: %v\n", pair)
+		}
+
+		v := strings.Split(pair, ":")
+		if len(v) > 2 {
+			return fmt.Errorf("Argument isn't a valid image:region pair: %v\n", pair)
+		}
+
+		image, region := v[0], v[1]
+		supportedRegions, err := API.ListRegions()
+		if err != nil {
+			return fmt.Errorf("could not list regions: %v\n", err)
+		}
+
+		var validRegion = false
+		for _, r := range supportedRegions {
+			if region == r {
+				validRegion = true
+				break
+			}
+		}
+
+		if validRegion == false {
+			return fmt.Errorf("%v is not a valid region\n", region)
+		}
```

Comment:
I think this would be much more ideal if we do a few things differently:

- call `API.ListRegions()` before the loop
    - There's no reason this output should change and thus no reason to hit the API on each loop iteration
- create a map (dictionary/hash) of supported regions (using a loop) outside of the main loop
- inside the main loop just check `supportedRegionMap[region]` to see if it's a valid region